### PR TITLE
[KIECLOUD-448] - Provide automated way to update product version on R…

### PR DESCRIPTION
**Pull request by spolti**
…HPAM and RHDM repositories

See: https://issues.redhat.com/browse/KIECLOUD-448

Signed-off-by: spolti <fspolti@redhat.com>

Thanks for submitting your Pull Request!

Please make sure your PR meets the following requirements:

- [x] Pull Request title is properly formatted: `[RHPAM-XYZ] Subject`
- [x] Pull Request contains link to the JIRA issue
- [x] Pull Request contains description of the issue
- [x] Pull Request does not include fixes for issues other than the main ticket
- [x] Attached commits represent units of work and are properly formatted
- [x] You have read and agreed to the Developer Certificate of Origin (DCO) (see `CONTRIBUTING.md`)
- [x] Every commit contains `Signed-off-by: Your Name <yourname@example.com>` - use `git commit -s`

### CODE REVIEW

File: scripts/update-version.py [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+import re
+import sys
+
+from ruamel.yaml import YAML
+
+# All rhpam modules that will be updated.
+MODULES = {"businesscentral", "businesscentral-monitoring", "controller",
+           "dashbuilder", "kieserver", "process-migration",
+           "smartrouter"}
+# modules are: module_name/modules/module_name
```
Comment (Kaitou786) [inline scripts/update-version.py:23]: Maybe specify full path as given in the example?
(module_name/modules/module_name/module.yaml)
And specify that the version will be updated in these files?

File: scripts/update-version.py [hunk 2]
```diff
@@ -0,0 +1,102 @@
+#!/usr/bin/python3
+# This script will to help to manage rhpam components modules version, it will update all needed files
+# Example of usage:
+#   # move the current version to the next one or rcX
+#   python scripts/update-version.py -v 7.15.1
```
Comment (Kaitou786) [inline scripts/update-version.py:5]: Maybe define `python3` as caller since till fedora 30 python2 is called when someone runs `python` command

**Review by Kaitou786 — CHANGES_REQUESTED** [review]
I think the regex needs to be tuned a bit, right now when I tested the script with command `python scripts/update-version.py -v 7.10.1 --confirm`. I can see the value also changed in the ports section. I could see it in `git diff`
```
ports:
 - value: 7.10
 ```

File: scripts/update-version.py (hunk 2 above)
Comment (spolti) [reply scripts/update-version.py:5]: +1.

File: scripts/update-version.py (hunk 1 above)
Comment (spolti) [reply scripts/update-version.py:23]: > I think the regex needs to be tuned a bit, right now when I tested the script with command `python scripts/update-version.py -v 7.10.1 --confirm`. I can see the value also changed in the ports section. I could see it in `git diff` in all module.yaml with value
> 
> ```
> ports:
>  - value: 7.10
> ```

Nice catch. updated.

Comment (spolti) [reply scripts/update-version.py:23]: do you mean improve the comment before this env?

**Review by Kaitou786 — DISMISSED** [review]
Looks good to me, optional nitpick just one thing not sure if it can be a problem or not, the script also changes the commented artifact name which is updated in our daily PRs, maybe ignore commented lines in the regex? See pic below:

Screenshot from 2021-01-05 13-48-42

**Comment by spolti** [comment]
> Looks good to me, optional nitpick just one thing not sure if it can be a problem or not, the script also changes the commented artifact name which is updated in our daily PRs, maybe ignore commented lines in the regex? See pic below:
> 
> Screenshot from 2021-01-05 13-48-42

Update the comment is also required so the cekit-cacher-bot can take care of it as well when updating the artifacts.

**Review by jakubschwan — COMMENTED** [review]
lgtm

**Comment by Kaitou786** [comment]
In the kieserver `module.yaml` I see one unwanted change(I guess):
Screenshot from 2021-01-12 18-01-51

Got this change when ran `python3 scripts/update-version.py -v 7.15.1 --confirm`

**Comment by spolti** [comment]
that's correct.
regex still needing to be tuned.

**Review by Kaitou786 — APPROVED** [review]
lgtm

**Pull request merged**